src/Lean/Server/FileWorker/RequestHandling.lean

```diff
@@ -237,7 +237,7 @@ open Parser.Command in
 partial def handleDocumentSymbol (_ : DocumentSymbolParams)
     : RequestM (RequestTask DocumentSymbolResult) := do
   let doc ← readDoc
-  asTask do
+  mapTask (← doc.cmdSnaps.waitHead?) fun _ => do
```

Comment:
Was this needed, given that we bind the `headerTask` before processing any request?

Reply:
You are right, I had already forgotten about that. Should we revert it or keep it just to be safe?

Reply:
For `RequestContext` specifically, we might also want to add a field for the header snapshot just to make this fact explicit, even if it is redundant with `doc.cmdSnaps`

Reply:
Stating this in the type system as you suggest is a good idea as the handlers don't need to do this kind of defensive programming then. We should also document that the header snap is duplicated in the request context.

Reply:
I didn't think this through, it can't replace `getLast!` in this case (and `cmdSnaps.getLastD headerSnap` would be plain weird IMO). I'd rather wait than introduce the field without a use case.